src/Compilers/Server/VBCSCompiler/ServerDispatcher.cs

```diff
@@ -350,6 +350,10 @@ internal static async Task<ConnectionData> HandleClientConnection(Task<IClientCo
                 // Unable to establish a connection with the client.  The client is responsible for
                 // handling this case.  Nothing else for us to do here.
                 CompilerServerLogger.LogException(ex, "Error creating client named pipe");
+                // It's possible for us to get persistent failures when attempting to handle a
+                // client connection, for example if the socket is in use somehow. We don't want to
+                // spin nonstop and tie up a core if this happens.
+                Thread.Sleep(100);
```

Comment:
This feels suspicious. This hasn't been needed up until this point. Similar bugs (spinning CPU) have been reported in the past by customers hence I don't think it's a case that we're missing. Why does Mono have this behavior here?

Reply:
The named pipe or whatever on my system was just in a hosed state I couldn't troubleshoot. Rebooting fixed it, I don't think it's a mono bug.

Reply:
I'd prefer to remove it. 

---
In reply to: [253151252](https://github.com/dotnet/roslyn/pull/31497#discussion_r253151252) [](ancestors = 253151252)